packages/router/router.ts

```diff
@@ -2554,26 +2564,31 @@ async function callLoaderOrAction(
         "Redirects returned/thrown from loaders/actions must have a Location header"
       );
 
-      // Support relative routing in redirects
-      let activeMatches = matches.slice(0, matches.indexOf(match) + 1);
-      let routePathnames = getPathContributingMatches(activeMatches).map(
-        (match) => match.pathnameBase
-      );
-      let requestPath = createURL(request.url).pathname;
-      let resolvedLocation = resolveTo(location, routePathnames, requestPath);
-      invariant(
-        createPath(resolvedLocation),
-        `Unable to resolve redirect location: ${result.headers.get("Location")}`
-      );
+      // Check if this an external redirect that goes to a new origin
+      let external = createURL(location).origin !== createURL("/").origin;
 
-      // Prepend the basename to the redirect location if we have one
-      if (basename) {
-        let path = resolvedLocation.pathname;
-        resolvedLocation.pathname =
-          path === "/" ? basename : joinPaths([basename, path]);
-      }
+      // Support relative routing in internal redirects
+      if (!external) {
```

Comment:
Don't do any post-processing of the redirect location if it's an external location

Reply:
I'm in the middle of migrating parts of a PHP website to remix, using nginx to route some urls to remix. In my particular case, if I `redirect()` from Remix to a PHP page, the origin would be the same, and this check would fail 😬 

Would testing for a scheme pattern (regex, or a lose check on `://`) instead negatively impact perfs?

Reply:
Hm, good call.  Is that something you can do in Remix today?  Or does it have this same issue?

Reply:
If we go with a manual check, I think we'd need to check for both protocol-less URLs (`//google.com/path`) as well as maybe a loose protocol check for `://`.  Perf shouldn't be a concern since this isn't a particularly hot code path - O(n) where n is the # of loaders/actions being run on a given navigation

Reply:
Remix has the same issue today.

I'm not sure we want to support protocol-less urls, as far as I know, those are only meant to apply the current scheme in *HTML*.

MDN says [relative or absolute](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location#directives), and RFC 9110 seems to require a `:` in [absolute URIs](https://www.rfc-editor.org/rfc/rfc3986.html#section-4.3).
*But that's a long document and my nerdiness is failing me 😅*

Also, if my `http://` remix app, is behind a `https://` nginx proxy, what scheme would `//my-domain/page` refer to? Doesn't seem that trivial from a server side perspective.

Reply:
ok yeah I figured it should since I stole the approach from there ;)

I haven't dug into the spec, but was more considering this from a browser-emulator perspective.  I did some quick testing and protocol-less URLs work in both of the redirect mechanisms at play here - neither of which are truly "handled" by RR/Remix - we're just handing off a location provided by the user to a built-in browser mechanism.  

Document redirects - returning a 301 with `Location: //google.com` works
Client-side redirects - `window.location.replace('//google.com')` works 

So I don't think we need to implement the logic on how to handle the redirect, so much as decide when to hand off the `Location` provided by the user directly to the browser instead of assuming it's a path inside our app.  If the browser is doing something not-technically-spec-compliant, we might be able to leave that as a browser concern?

Reply:
Done in https://github.com/remix-run/react-router/issues/9634